### ISSUE TITLE
EA2 getter functions can return wrong data or datatype

### ISSUE THREAD

**Issue by Doubleumc** [nrlulz/ACF#189]
Several of the EA2 ACF getter functions do not return the expected data, and some don't return the expected datatype.  This makes using EA2 for ACF difficult as some of this data is not available by wirelink, it is only available through these functions.  Said functions can be split into three groups:

- Getters always returning false or 0, with correct datatype:
 - acfActive, acfInPowerband, acfIsFuel, acfFuelRequired, acfRefuelDuty, acfFuel, acfFuelUse, acfPeakFuelUse, acfGear, acfNumGears, acfFinalRatio, acfTotalRatio, acfIsDual, acfInGear, acfGearRatio, acfTorqueOut, acfSpread, acfIsReloading, acfMagRounds, acfAmmoCount, acfTotalAmmoCount, acfPropHealth, acfPropArmor, acfPropHealthMax, acfPropArmorMax, acfPropDuctility, acfHitClip, acfIsAmmo, acfRounds, acfRoundType, acfAmmoType, acfCaliber, acfMuzzleVel, acfProjectileMass, acfFLSpikes, acfFLSpikeMass, acfFLSpikeRadius, acfPenetration, acfBlastRadius
- Getters always returning true, with correct datatype:
 - acfInfoRestricted
- Getters always returning false, when a number is expected:
 - acfRPM, acfTorque, acfPower, acfThrottle

As far as I know this is a complete list -- all other functions worked as expected.

This bug presents itself fully in multiplayer, and I have tested it on several servers:
* 70.42.74.154:27015 -- [FRS] ACF Combat, Engineering, Building
* 192.157.56.114:27017 -- GGG - Toshkent - Strength in Justice
* 45.56.78.184:27015 -- DildioBuildo - Stay Straight
* 66.151.244.118:27015 -- TCW - Smooth Jazz - BUILD ...
* 72.5.102.29:27015 -- Malleus Gaming - Sandbox ...

This bug also presents itself partially in singleplayer.  acfInfoRestricted still returns true, and when passed an invalid entity the "number is expected" group still returns false.  All other functions worked as expected.

Steps to reproduce:
1. Connect to a multiplayer server with both ACF and EA2 enabled.
2. Spawn an ACF entitiy -- for this example I am using an engine.
3. Spawn an EA2 chip with the following code:
```
server {
    input entity Engine
    event trigger() {
        Engine.acfActive(true)
        print("This should be true: ", Engine.acfActive())
        print("This should be a number: ", Engine.acfRPM())
    }
}
```
Then simply link the EA2 to the engine.

With the above code what we expect to see is the following message:
```
This should be true: <boolean: true>
This should be a number: 900

```

Instead, the message we receive is this:
```
This should be true: <boolean: false>
This should be a number: false

```

**Comment by Bubbus** [comment]
I got this problem too, it occurs if "sbox_acf_restrictinfo 1" on the server.  It seems that the EA2 lib forgets not to restrict you from your own ents.

I can look into it when I get time but I can't guarantee anything.  Might be better for FreeFry to check it out as I think he wrote it.

**Comment by Bubbus** [comment]
Ok I think i fixed it in c2a9ac435f5ea6c2d79ab99037e2a696a870e2f3, lemme know if it works your way.  In the meantime as a compatibility patch, use an E2 like this and wire it to some EA2 inputs:
```
@name ea2 acf compat patch
@inputs Gun:entity
@outputs GunOut:entity AmmoLeft AmmoType:string

if (Gun)
{
    GunOut = Gun
    AmmoLeft = Gun:acfAmmoCount()
    AmmoType = Gun:acfAmmoType()
}
else
{
    GunOut = noentity()
    AmmoLeft = 0
    AmmoType = ""
}

interval(100)

```
